lib/galaxy/tools/parameters/basic.py

```diff
@@ -1123,16 +1122,16 @@ def get_column_list(self, trans, other_values):
             # Use representative dataset if a dataset collection is parsed
             if isinstance(dataset, trans.app.model.HistoryDatasetCollectionAssociation):
                 dataset = dataset.to_hda_representative()
-            # Columns can only be identified if metadata is available
-            if not hasattr(dataset, 'metadata') or not hasattr(dataset.metadata, 'columns'):
+            # Columns can only be identified if the dataset is ready and metadata is available
+            if not hasattr(dataset, 'state') or \
+                dataset.state != galaxy.model.Dataset.states.OK or \
+                not hasattr(dataset, 'metadata') or \
+                not hasattr(dataset.metadata, 'columns') or \
+                not dataset.metadata.columns:
```

Comment:
You need to add one extra indentation to lines 1127-1130 to make flake8 happy.

Reply:
Thx. Will do.